### PR TITLE
LitElement conversion part 4: API docs#65 - [skip ci]

### LINKED ISSUE #65: LitElement conversion part 4: API docs
[Example PR](https://github.com/vaadin/vaadin-split-layout/pull/141).

### Document API

- Add JSDoc comment for the custom element definition:

  - Use markdown for a short description.

  - Use `@csspart` JSDoc annotation for documenting Shadow Parts:

    - [`part="value"`](https://github.com/vaadin/vaadin-progress-bar/blob/893d5ef2c2f5c2f91ea04dbbad9b2435f51cf3ef/src/vaadin-progress-bar.html#L65)
    - [`part="bar"`](https://github.com/vaadin/vaadin-progress-bar/blob/893d5ef2c2f5c2f91ea04dbbad9b2435f51cf3ef/src/vaadin-progress-bar.html#L66)
  
- Add JSDoc comments for all the public properties
  
  - [`value`](https://github.com/vaadin/vaadin-progress-bar/blob/893d5ef2c2f5c2f91ea04dbbad9b2435f51cf3ef/src/vaadin-progress-mixin.html#L21-L23)
  - [`min`](https://github.com/vaadin/vaadin-progress-bar/blob/893d5ef2c2f5c2f91ea04dbbad9b2435f51cf3ef/src/vaadin-progress-mixin.html#L28-L30)
  - [`max`](https://github.com/vaadin/vaadin-progress-bar/blob/893d5ef2c2f5c2f91ea04dbbad9b2435f51cf3ef/src/vaadin-progress-mixin.html#L36-L38)
  - [`indeterminate`](https://github.com/vaadin/vaadin-progress-bar/blob/893d5ef2c2f5c2f91ea04dbbad9b2435f51cf3ef/src/vaadin-progress-mixin.html#L44-L47)

### Prepare docs

- Copy `index.html` file from skeleton, update title and import the component

- Inside `api-viewer`, add `template` for `dir` attribute. [Example](https://github.com/vaadin/vaadin-radio-button/blob/1b274fba6d59238876e25c43e8a0400811a92e9b/index.html#L38-L44) 

- Inside `api-viewer`, add `template` for `theme` (contrast, error, success). [Example](https://github.com/vaadin/vaadin-split-layout/blob/6b70d8149f5392583ac866a1c237e4ac2bace6b4/index.html#L30-L36) 

### Serve docs

- Run `npm start` to run server for browsing API docs and demo

- Open page, go to `Demo` and go to `Knobs` tab

- Try all the knobs (value, dir, theme etc) and test that they work

### Create dist

- Copy `rollup.config.js` from skeleton (used to build CDN demo)

- Run the npm script to test how the built demo works locally:

```sh
$ npm run dist && npm run serve:dist
```

Note: use `[skip ci]` in the commit message.

### PR DESCRIPTION
**Pull request by phanfullstack-12345**
Fixes #65

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -2,13 +2,46 @@
...
+      @media(max-width: 799px) {
+        api-viewer {
+          margin: 0;
+        }
+      }
+    </style>
   </head>
   <body>
-    <iron-component-page src="vaadin-progress-bar.html"></iron-component-page>
+    <api-viewer src="custom-elements.json" exclude-knobs="autofocus,items,tabIndex">
```
Comment (web-padawan) [inline index.html]: Please remove `exclude-knobs`. This [feature](https://github.com/web-padawan/api-viewer-element#exclude-knobs) is only needed to hide certain properties.
Progress-bar doesn't have those.

File: index.html [hunk 2]
```diff
@@ -2,13 +2,46 @@
...
+      <template data-element="vaadin-progress-bar" data-target="knob" data-attr="dir" data-type="select">
+        <select>
+          <option value=""></option>
+          <option value="ltr"></option>
+          <option value="rtl"></option>
+        </select>
+      </template>
+      <template data-element="vaadin-progress-bar" data-target="knob" data-attr="theme" data-type="select">
+        <select>
+          <option value="contrast"></option>
```
Comment (web-padawan) [inline index.html:39]: Please add empty option: `<option value=""></option>` to allow deselect `theme` value.

File: src/vaadin-progress-bar.ts [hunk 3]
```diff
@@ -2,6 +2,17 @@ import { customElement } from 'lit-element';
...
+ * `<vaadin-progress-bar>` is a Web Component for progress bars.
+ * Content can be any HTML, including nested `<vaadin-progress-bar`.
+ *
+ * @csspart part="bar" - Progress-bar's background
+ * @csspart part="value" - Progress-bar's foreground
+ *
+ * See [ThemableMixin – how to apply styles for shadow parts](https://github.com/vaadin/vaadin-themable-mixin/wiki)
+ *
+ */
+
```
Comment (web-padawan) [inline src/vaadin-progress-bar.ts]: Please remove this empty newline.

File: src/vaadin-progress-bar.ts [hunk 4]
```diff
@@ -2,6 +2,17 @@ import { customElement } from 'lit-element';
 import { ProgressBarBase } from './vaadin-progress-bar-base';
 import { ProgressMixin } from './vaadin-progress-mixin';
 
+/**
+ * `<vaadin-progress-bar>` is a Web Component for progress bars.
+ * Content can be any HTML, including nested `<vaadin-progress-bar`.
```
Comment (web-padawan) [inline src/vaadin-progress-bar.ts]: Please remove this line. It is a copy paste from `vaadin-spit-layout`, and the statement isn't true for `vaadin-progress-bar` (we don't expect it to have any content).

File: src/vaadin-progress-bar.ts [hunk 5]
```diff
@@ -2,6 +2,17 @@ import { customElement } from 'lit-element';
...
+ *
+ * @csspart part="bar" - Progress-bar's background
+ * @csspart part="value" - Progress-bar's foreground
+ *
+ * See [ThemableMixin – how to apply styles for shadow parts](https://github.com/vaadin/vaadin-themable-mixin/wiki)
```
Comment (web-padawan) [inline src/vaadin-progress-bar.ts]: Please remove `ThemableMixin`, we don't mention it in the docs for next generation components.

**Review by web-padawan — CHANGES_REQUESTED** [review]

File: index.html (hunk 1 above)
Comment (phanfullstack-12345) [reply index.html]: done

File: index.html (hunk 2 above)
Comment (phanfullstack-12345) [reply index.html:39]: done

File: src/vaadin-progress-bar.ts (hunk 3 above)
Comment (phanfullstack-12345) [reply src/vaadin-progress-bar.ts]: done

File: src/vaadin-progress-bar.ts (hunk 4 above)
Comment (phanfullstack-12345) [reply src/vaadin-progress-bar.ts]: done

File: src/vaadin-progress-bar.ts (hunk 5 above)
Comment (phanfullstack-12345) [reply src/vaadin-progress-bar.ts]: done

**Review by web-padawan — APPROVED** [review]

**Pull request merged**